src/helpers/gpt.py

```diff
@@ -213,17 +253,26 @@ async def process_batch_async(self, session, batch, custom_guidelines_prompt=Non
         for formatted_prompt in batch:
             # formatted_prompt = custom_guidelines_prompt.format(prompt) if custom_guidelines_prompt else self.GUIDELINES_PROMPT_TEMPLATE.format(prompt)
             # cache_id = f"{self.prompt_id}: {formatted_prompt}"
-
-            # TODO: hash it?
-            prompt_id = formatted_prompt
+            is_message_list = False
+            if isinstance(formatted_prompt, list):
+                # If the prompt is a list, convert it to a string representation
+                prompt_id = self.stringify_messages(formatted_prompt)
+                is_message_list = True
+            else:
+                # TODO: hash it?
+                prompt_id = formatted_prompt
             input_token_count = self.count_tokens(formatted_prompt)
 
             if prompt_id in self.cache:
                 self.token_usage["cached_input_tokens"] += input_token_count  # Cached input tokens
                 responses.append(self.cache[prompt_id])
+                print(f"Using cached response for prompt ID: {prompt_id}")
             else:
                 self.token_usage["input_tokens"] += input_token_count  # Normal input tokens
-                task = asyncio.create_task(self.make_openai_request_async(session, formatted_prompt))
+                if is_message_list:
+                    task = asyncio.create_task(self.make_openai_request_async(session, messages_list=formatted_prompt))
+                else:
+                    task = asyncio.create_task(self.make_openai_request_async(session, formatted_prompt=formatted_prompt))
```

Comment:
@dayeonki Do you want to also take a quick look at this file's changes and make sure it looks good to you too?